presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+
+        checkArgument(start >= 0, "Start offset %s for column '%s' must be greater or equal 0", start, columnHandle.getName());
+        end.ifPresent(endValue -> {
+            checkArgument(endValue >= 0, "End offset %s for column '%s' must be greater or equal 0", endValue, columnHandle.getName());
+            checkArgument(endValue > start, "End offset %s for column '%s' must greater than start offset", endValue, columnHandle.getName());
+        });
+
+        int length = end.isPresent() ? end.getAsInt() - start : fieldType.getSize();
+
+        if (!isVarcharType(columnHandle.getType())) {
+            checkArgument(!end.isPresent() || end.getAsInt() - start == length,
+                    "Bytes mapping for column '%s' does not match dataFormat '%s'; expected %s bytes but got %s",
+                    columnHandle.getName(),
+                    length,
+                    end.getAsInt() - start);
+        }
+
+        return length;
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        buffer.put(new byte[valueLengths[currentColumnIndex]], currentBufferPosition, valueLengths[currentColumnIndex]);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        buffer.putLong(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        buffer.putInt(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        buffer.putShort(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        buffer.put(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        buffer.putDouble(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        buffer.putFloat(currentBufferPosition, value);
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        buffer.put(currentBufferPosition, (byte) (value ? 1 : 0));
+        currentBufferPosition += valueLengths[currentColumnIndex];
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        buffer.put(value.getBytes(StandardCharsets.UTF_8), currentBufferPosition, valueLengths[currentColumnIndex]);
```

Comment:
this can throw `value.getBytes()` may return shorter array than `valueLengths[currentColumnIndex]`

Reply:
the best solution I can think of is create an intermediate `ByteBuffer` of the correct length for that column and then put the array from that into the actual `ByteBuffer` like this
```java
protected void appendString(String value)
{
    ByteBuffer tempBuf = ByteBuffer.allocate(valueLengths[currentColumnIndex]);
    tempBuf.put(value.getBytes(StandardCharsets.UTF_8), 0, valueLengths[currentColumnIndex]);
    buffer.put(tempBuf.array(), currentBufferPosition, valueLengths[currentColumnIndex]);
    currentBufferPosition += valueLengths[currentColumnIndex];
}
```

Reply:
Yes for now I think that would be best. In the future it would be a good idea to rethink how the raw encoder maps values and better define that in the documentation.

Reply:
Actually I meant that the code will throw if `String value` is shorter than `valueLengths[currentColumnIndex]`. It is still the case with updated code.
`buffer.put(srcArray, offset, length)` requires that there is at least `length` bytes in `srcArray` starting from `offset`.

Also I have just noticed that you pass `currentBufferPosition` source offset. This does feels wrong as the offset should be in source string space. And `currentBufferPosition` represents position in target buffer. You should pass 0 as offset i think.

Reply:
> Actually I meant that the code will throw if `String value` is shorter than `valueLengths[currentColumnIndex]`
In this case I think that it is best if we throw an exception for now, otherwise the decoder will just read garbage.
> Also I have just noticed that you pass `currentBufferPosition` source offset. This does feels wrong as the offset should be in source string space.
I noticed this as well while working on this issue last week. I also realized that `currentBufferPosition` is a redundant variable because I can just use the built in `position` feature of the `ByteBuffer` to track the position and then reset it in `toByteArray`.

Reply:
>> Actually I meant that the code will throw if String value is shorter than valueLengths[currentColumnIndex]

> In this case I think that it is best if we throw an exception for now, otherwise the decoder will just read garbage.

Yeah. I think I agree. Unless we want to pad with some whitespace. But still it should be a descriptive exception. Not IndexOutOfBounds... :)

Reply:
what if it's `IndexOutOfBoundsException` with a descriptive message?

Reply:
That should be fine :)